functions/createNewOrgAPI.js

```diff
@@ -96,27 +96,46 @@ exports.createNewTeamWithLabels = functions.https.onRequest((request, response)
         const statusLabels = data.StatusLabels;
         const priorityLabels = data.PriorityLabels;
         const difficultyLabels = data.DifficultyLabels;
+        const appKey = data.AppKey;
         let orgId;
+        let orgDomain;
 
         const promise1 = db.collection("Organizations").where("OrganizationDomain", "==", organizationDomain).get().then((querySnapshot) => {
             querySnapshot.forEach((doc) => {
                 orgId = doc.data().OrganizationId;
             });
-            const p1 = db.collection("Organizations").doc(organizationDomain).collection("Teams").doc(teamName).set({
-                TeamName: teamName,
-                TeamDescription: teamDescription,
-                TeamManagerEmail: teamManagerEmail,
-                TeamMembers: teamMembers,
-                TaskLabels: taskLabels,
-                StatusLabels: statusLabels,
-                PriorityLabels: priorityLabels,
-                DifficultyLabels: difficultyLabels,
-                TotalTeamTasks: 0,
-                CurrentSprintId: 0,
-                OrganizationId: orgId,
-                TeamId: teamId,
+            db.collection("Organizations").where("AppKey", "==", appKey).get().then((org)=>{
+                org.forEach((doc)=>{
+                    orgDomain = doc.data().OrganizationDomain;
+                });
```

Comment:
The logic here is not correct, you need to get the orgDomain above and then get the team, if team exists, then update else set.